engine/src/engine.rs

```diff
@@ -1915,6 +1920,23 @@ impl<'env, I: IO + Copy, E: Env, M: ModExpAlgorithm> Backend for Engine<'env, I,
         result
     }
 
+    fn is_empty_storage(&self, address: H160) -> bool {
+        let address = Address::new(address);
+        if self
+            .contract_storage_cache
+            .borrow()
+            .contains_key(&(address, H256::zero()))
+        {
+            return true;
+        }
```

Comment:
I don't think this implementation is correct. If the storage cache contains a key then it is not empty (should return `false`). Also what if `0x00` is not used, but some other slot is?

Reply:
In that particular case, we're just checking a storage cache.
But if it doesn't exist we'll just read storage by the key: (address, index), where index = 0;
Index zero should exist, as it's the impossible case in EVM to have index > 0, but don't have an index = 0;

Also, we can't store storage keys for an account, as it'll cost NEAR storage.

Reply:
> it's the impossible case in EVM to have index > 0, but don't have an index = 0

Why is this the case? I thought you could write a value to any slot leaving the others unused. The storage usage convention in Solidity contracts is to store values for fields in the contract starting at index 0, but for arbitrary contracts written in EVM bytecode directly that does not need to be the case.

Reply:
hm... it's problematic to get account storage status on `aurora-engine`. We don't know the account storage keys (indexes) for current EVM accounts. I believe it's the optimal solution for now to assume that `inxed=0` exists.
In the future, we can store account empty storage flag.

Reply:
Yeah, we'll need to think about a more robust solution to this problem. A flag is a nice idea. It'll require a storage migration for all existing accounts, but it's probably worth the effort to do this.